### DISCUSSION TITLE
Render too many markers problem

### DISCUSSION THREAD

**Discussion by nikitassh** [visgl/react-google-maps#151] [Q&A]
I used example https://github.com/leighhalliday/next-google-maps/tree/main, and i'm trying to render 18.000 points on map, but render time takes at least 3 min.

**Comment by usefulthink** [comment]
From my experience, regular markers are good for rendering something like 100 points on the map, with advanced markers I wouldn't go beyond maybe 1000 points. If you're in the region of over 10k points, you should probably have a look at deck.gl for example, which will handle that amount with ease.

(Note to self: we should add something about that to the documentation)

**Reply by nikitassh** [reply]
Could you provide some example code, where you are using deckgl, google maps and marker-clustering.

**Reply by usefulthink** [reply]
Sorry, missed your reply. Here's an example showing how Markerclustering can be done with deck.gl: https://deck.gl/examples/icon-layer ([source](https://github.com/visgl/deck.gl/tree/8.9-release/examples/website/icon)). And here is a guide about using deck.gl together with this library: https://visgl.github.io/react-google-maps/docs/guides/deckgl-integration (I'd recommend you try the `GoogleMapsOverlay` version from that page first).
